feat(media): add session switching functionality

Dusk-afk opened the pull request:
This pull request adds session switching functionality to the media widget, addressing #139.
Using scroll wheel on the media widget, sessions can be switched.
- Scroll down = Next session
- Scroll up = Previous session

#### Changes:
* Added `_current_session_only` decorator to ensure session update functions are called only if the session is the same as the current session.
* Added session switching logic with `switch_session` method and updated `play_pause`, `prev`, and `next` methods to use the current session.
* Added logic to hide the thumbnail if there is no media in the session in `_on_media_properties_changed`.
* Implemented `wheelEvent` to switch media sessions using the mouse wheel.

#### Demo:
Following is a demo of the media widget handling 3 sessions:
1. Apple Music
2. Spotify
3. YT Music (Browser)

https://github.com/user-attachments/assets/7e8a80ab-4bb7-4ff6-9539-10d1e6f0e73d

Video-Nomad commented:
Thanks for the PR! Works really well.
Two minor things I've noticed so far:
1. When I pause the video on YouTube and then close the tab it just gets stuck on that empty session. Is it possible to switch to the next active session if that happens?

https://github.com/user-attachments/assets/79f36ee8-cdc8-4311-8b77-75bfd4220bf8

2. Just some minor logging spam that's unnecessary unless it's DEBUG=True

![image](https://github.com/user-attachments/assets/34c0fa55-dd16-4176-9694-48c29c14f03b)

Dusk-afk commented:
Thanks for the feedback! Glad to hear it’s working well. I've addressed both points:
1. Implemented a check to detect when the media info is empty and the current session is closed. If this condition is met, the session will automatically switch.
2. Added a condition to log only when `DEBUG` is enabled.
